tests/unit/reports/multi_table/_properties/test_relationship_validity.py

```diff
@@ -0,0 +1,302 @@
+                    'parent_primary_key': 'col1',
+                    'child_table_name': 'table2',
+                    'child_foreign_key': 'col6'
+                },
+                {
+                    'parent_table_name': 'table1',
+                    'parent_primary_key': 'col1',
+                    'child_table_name': 'table3',
+                    'child_foreign_key': 'col7'
+                },
+                {
+                    'parent_table_name': 'table2',
+                    'parent_primary_key': 'col6',
+                    'child_table_name': 'table4',
+                    'child_foreign_key': 'col8'
+                },
+            ]
+        }
+        relationship_validity = RelationshipValidity()
+
+        # Run
+        num_iterations = relationship_validity._get_num_iterations(metadata)
+
+        # Assert
+        assert num_iterations == 3
+
+    @patch('sdmetrics.reports.multi_table._properties.relationship_validity.'
+           'CardinalityBoundaryAdherence')
+    @patch('sdmetrics.reports.multi_table._properties.relationship_validity.ReferentialIntegrity')
+    def test_get_score(
+        self, mock_referentialintegrity, mock_cardinalityboundaryadherence,
+        real_data_fixture, synthetic_data_fixture, metadata_fixture
+    ):
+        """Test the ``get_score`` function.
+
+        Test that when given a ``progress_bar`` and relationships, this calls
+        ``CardinalityBoundaryAdherence`` and ``ReferentialIntegrity`` compute
+        method for each relationship.
+        """
+        # Setup
+        mock_referentialintegrity.compute.return_value = 0.7
+        mock_referentialintegrity.__name__ = 'ReferentialIntegrity'
+        mock_cardinalityboundaryadherence.compute.return_value = 0.3
+        mock_cardinalityboundaryadherence.__name__ = 'CardinalityBoundaryAdherence'
+        mock_compute_average = Mock(return_value=0.5)
+        relationship_validity = RelationshipValidity()
+        relationship_validity._compute_average = mock_compute_average
+        progress_bar = Mock()
+
+        real_data = real_data_fixture
+        synthetic_data = synthetic_data_fixture
+        metadata = metadata_fixture
+
+        # Run
+        score = relationship_validity.get_score(
+            real_data=real_data, synthetic_data=synthetic_data, metadata=metadata,
+            progress_bar=progress_bar
+        )
+
+        # Assert
+        expected_details_property = pd.DataFrame({
+            'Parent Table': ['users', 'users'],
+            'Child Table': ['sessions', 'sessions'],
+            'Primary Key': ['user_id', 'user_id'],
+            'Foreign Key': ['user_id', 'user_id'],
+            'Metric': ['ReferentialIntegrity', 'CardinalityBoundaryAdherence'],
+            'Score': [0.7, 0.3],
+        })
+
+        assert score == 0.5
+        progress_bar.update.assert_called()
+        assert progress_bar.update.call_count == 1
+        mock_compute_average.assert_called_once_with()
```

Comment:
minor: if it's empty I think you can do
```
mock_compute_average.assert_called_once()
```

Reply:
Yes, done in [a5afedd](https://github.com/sdv-dev/SDMetrics/pull/488/commits/a5afeddfa288b600e992ca10294b44f622cec736)